provider/coredns/coredns.go

```diff
@@ -204,15 +204,24 @@ func loadRoots(caPath string) (*x509.CertPool, error) {
 
 // builds etcd client config depending on connection scheme and TLS parameters
 func getETCDConfig() (*etcdcv3.Config, error) {
+    config := etcdcv3.Config{}
 	etcdURLsStr := os.Getenv("ETCD_URLS")
 	if etcdURLsStr == "" {
 		etcdURLsStr = "http://localhost:2379"
 	}
 	etcdURLs := strings.Split(etcdURLsStr, ",")
 	firstURL := strings.ToLower(etcdURLs[0])
+    etcdUser, etcdUserDefined := os.LookupEnv("ETCD_USER")
+    if etcdUserDefined {
+        config.Username = etcdUser
+    }
+    etcdPass, etcdPassDefined := os.LookupEnv("ETCD_USER")
```

Comment:
I think it should look like this:
```suggestion
    etcdPass, etcdPassDefined := os.LookupEnv("ETCD_PASS")
```